lib/Sema/ConstraintSystem.cpp

```diff
@@ -5275,6 +5275,12 @@ void constraints::simplifyLocator(ASTNode &anchor,
       break;
     }
 
+    case ConstraintLocator::AnyPatternDecl: {
+      anchor = anchor.get<Pattern *>();
+      path = path.slice(1);
+      break;
```

Comment:
I don’t think this is correct. `anchor = anchor.get<Pattern *>();` is a no-op but you’re slicing away the path element regardless. I would assume this locator can’t be simplified.

Reply:
Hm, the idea was to make sure that the anchor is indeed a pattern and drop the element since pattern represents the decl in this case…

Reply:
Ah, I see. I think that makes sense. So the `AnyPatternDecl` constraint locator is basically just a marker that the pattern that’s the anchor is an `AnyPattern`. In that case I’ve got a couple suggestions
- Could you add the discussion we had here in a doc comment
- Change the `anchor` reassignment to `assert(anchor.is<Pattern *>())`
- Shouldn’t we check that the anchor is an `AnyPattern`, i.e. `assert(isPattern<AnyPattern>(anchor))` (or something along those lines.

Reply:
Will do!

Reply:
Done!